src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -132,11 +133,25 @@ public static partial class JsonSerializer
                         clearMax = bytesInBuffer;
                     }
 
+                    int start = 0;
```

Comment:
nit: rather than having this local, create a Span<byte> local that points to the buffer:
```C#
var inputSpan = new Span<byte>(buffer, 0, bytesInBuffer);
```

Then, in the `if (loopCount == 0)` check below, we can do:
```C
inputSpan = inputSpan.Slice(utf8BomLength);
```

Reply:
Also, if you are making this change, I would make it a `ReadOnlySpan<byte>` rather than a read-write Span. I know it exists as a `Span<byte>` today, but there is no reason for it to not be readonly.

cc @steveharter

Reply:
So the suggested change will involve 1 new object and 1 slice call. Should we think about performance here? I don't have a strong preference either way

Reply:
ReadOnlySpan is not an "object" (in the typical/heap allocation sense) and is quite cheap. Same with the slice operation.

Also, we are already creating the span when calling the ReadCore method. I was mainly suggesting avoiding making it a local and avoid the slicing/passing in start index in the common case (where there is no bom).

Reply:
Ah, this is an async method, so no parameter or local spans allowed. I'd have a to create a local function and pass in the bytesInBuffer and the other locals as `ref`. Doesn't seem worth it to me? Maybe we should leave it as it is? I did make the change to a `ReadOnlySpan` though.

Reply:
Good point! I keep forgetting :)

Leaving it as is fine (not that we can help it anyway).